gfx.h

```diff
@@ -6280,7 +6280,7 @@ class GfxInternal
             }
             state_object_properties->Release();
         }
-        if(kernel.vertex_stride_ > 0)
+        if(!is_compute)
```

Comment:
Any reason for reverting this change? This condition had to be changed to `if(kernel.vertex_stride_ > 0)` due to the addition of mesh shaders: https://github.com/gboisse/gfx/pull/81.

Reply:
As I implemented manual vertex fetch in the vertex shader for skinned mesh, vertex_stride_ can be 0 although we still should set index buffer via IASetIndexBuffer. But, given that there are mesh shaders, I think it should be further adjusted.

Reply:
I see.

So, we now have a proper check available to query whether a kernel uses standard graphics pipeline:
https://github.com/gboisse/gfx/blob/77ac4c63495ff0f6ad3dab8075c1c3e7beba9ef0/gfx.h#L1082

Think this check should therefore be `if(kernel.isGraphics())`, which would ensure that the IB is being set if there's one (even without a vertex buffer) and would skip these calls when using a mesh shader-based pipeline.

Reply:
btw, bit unrelated but I recently went and chased down all instances of unreleased D3D12 resources that we had and realised that this DXR-1.0 code was leaking handles every single frame (well, every dispatch to be more precise):
https://github.com/gboisse/gfx/blob/77ac4c63495ff0f6ad3dab8075c1c3e7beba9ef0/gfx.h#L6216-L6217

So FYI, `QueryInterface()` does "acquire" the handle meaning you do need to release it afterwards:
https://github.com/gboisse/gfx/blob/77ac4c63495ff0f6ad3dab8075c1c3e7beba9ef0/gfx.h#L6281

Reply:
Done.